src/ngraph/op/arg_minmax.cpp

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+* Copyright 2017-2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "ngraph/op/arg_minmax.hpp"
+#include "ngraph/op/broadcast.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+op::IndexReduction::IndexReduction(const std::string& node_type,
+                                   const std::shared_ptr<Node>& arg,
+                                   size_t axis,
+                                   bool keep_dimensions,
+                                   bool is_int64)
```

Comment:
This is the first op to have an opinion on what type an index should have. Morally it seems to me like `uint64` is the right choice since we usually use `size_t` for stuff like this in nGraph, but I suppose if we go with that we'll be forcing an extra conversion onto the bridges that backends will have to optimize away, so... oh well.

Will int32 and int64 suffice? (I know those are all the options where TensorFlow is concerned.)

I think maybe we should pass this in as an `element::Type` parameter and check that it's in the allowed set, rather than pass a bool here. That way if we change our minds later and need to support argmax/argmin on a wider set of output types there won't be as big an issue where backwards-compat is concerned.